mongomock/collection.py

```diff
@@ -595,6 +595,59 @@ def _update(self, spec, document, upsert=False, manipulate=False,
                             else:
                                 push_results.append(value)
 
+                            subdocument[nested_field_list[-1]] = push_results
+                elif k == '$pushAll':
+                    for field, value in iteritems(v):
```

Comment:
I have the feeling that most of this could be factorized with the $push implementation.